rosetta/services/block.go

```diff
@@ -0,0 +1,730 @@
+package services
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+
+	"github.com/coinbase/rosetta-sdk-go/server"
+	"github.com/coinbase/rosetta-sdk-go/types"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/harmony-one/harmony/core/rawdb"
+	hmytypes "github.com/harmony-one/harmony/core/types"
+	"github.com/harmony-one/harmony/hmy"
+	internalCommon "github.com/harmony-one/harmony/internal/common"
+	"github.com/harmony-one/harmony/internal/utils"
+	"github.com/harmony-one/harmony/rosetta/common"
+	"github.com/harmony-one/harmony/rpc"
+	rpcV2 "github.com/harmony-one/harmony/rpc/v2"
+	"github.com/harmony-one/harmony/staking"
+	stakingTypes "github.com/harmony-one/harmony/staking/types"
+)
+
+// BlockAPIService implements the server.BlockAPIServicer interface.
+type BlockAPIService struct {
+	hmy *hmy.Harmony
+}
+
+// NewBlockAPIService creates a new instance of a BlockAPIService.
+func NewBlockAPIService(hmy *hmy.Harmony) server.BlockAPIServicer {
+	return &BlockAPIService{
+		hmy: hmy,
+	}
+}
+
+// Block implements the /block endpoint
+func (s *BlockAPIService) Block(
+	ctx context.Context, request *types.BlockRequest,
+) (response *types.BlockResponse, rosettaError *types.Error) {
+	if err := assertValidNetworkIdentifier(request.NetworkIdentifier, s.hmy.ShardID); err != nil {
+		return nil, err
+	}
+
+	var block *hmytypes.Block
+	var currBlockID, prevBlockID *types.BlockIdentifier
+	if block, rosettaError = s.getBlock(ctx, request); rosettaError != nil {
+		return nil, rosettaError
+	}
+	currBlockID = &types.BlockIdentifier{
+		Index: block.Number().Int64(),
```

Comment:
Yea it should be, however, the type requires an int64. At some point, rosetta took in -1 to indicate genesis blocks. They have since changed the spec, maybe the SDK will receive an update.